src/python/pants/engine/internals/engine_test.py

```diff
@@ -589,3 +592,30 @@ def test_process_digests_on_workunits(self):
         byte_outputs = self._scheduler.digests_to_bytes([stdout_digest, stderr_digest])
         assert byte_outputs[0] == result.stdout
         assert byte_outputs[1] == result.stderr
+
+    def test_context_object(self):
+        self._init_engine()  # need to call this so that self._scheduler is not None when we pass it to StreamingWorkunitHandler
+
+        def callback(workunits, **kwargs) -> None:
+            context = kwargs["context"]
+            assert type(context) == StreamingWorkunitContext
```

Comment:
I suggest not using an assert here, use isinstance (instead of `type(context)== StreamingWorkunitContext `) and raise an exception (TypeError) if the check fails.

Reply:
ah. nevermind, this is a test, an assert is appropriate here. just use isinstance then.